### PR TITLE
chore: Update EKS module version to `v20.0`

### PR DESCRIPTION
**Pull request by bryantbiggs**
# Description

- Update EKS module version to `v20.0`
	- There are a few patterns that are still on v19 and will require additional changes - saving those for separate PR(s)

### Motivation and Context

- Updating to latest module

### How was this change tested?

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

### CODE REVIEW

File: .github/workflows/e2e-parallel-destroy.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
       - name: Setup Terraform
         uses: hashicorp/setup-terraform@v3
         with:
-          terraform_version: 1.0.0
+          terraform_version: 1.3.10
```
Comment (bryantbiggs) [inline .github/workflows/e2e-parallel-destroy.yml:61]: The EKS module now requires MSTV of 1.3 so updating this in our e2e tests as well

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -29,6 +29,7 @@ repos:
           - '--args=--only=terraform_naming_convention'
           - '--args=--only=terraform_required_version'
           - '--args=--only=terraform_required_providers'
+          - '--args=--only=terraform_unused_required_providers'
```
Comment (bryantbiggs) [inline .pre-commit-config.yaml:32]: nifty little lint check to make sure we remove providers when they aren't being utilized

File: patterns/elastic-fabric-adapter/main.tf [hunk 3]
```diff
@@ -127,29 +127,6 @@ module "eks" {
         group_name = aws_placement_group.efa.name
       }
 
-      pre_bootstrap_user_data = <<-EOT
```
Comment (bryantbiggs) [inline patterns/elastic-fabric-adapter/main.tf:130]: this is available by default today through 1.25+ and will soon be available through 1.23+

File: patterns/elastic-fabric-adapter/main.tf [hunk 4]
```diff
@@ -275,7 +252,7 @@ resource "kubernetes_daemonset" "aws_efa_k8s_device_plugin" {
 
         container {
           name  = "aws-efa-k8s-device-plugin"
-          image = "602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/aws-efa-k8s-device-plugin:v0.3.3"
+          image = "602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/aws-efa-k8s-device-plugin:v0.4.3"
```
Comment (bryantbiggs) [inline patterns/elastic-fabric-adapter/main.tf:255]: this was updated recently for some CVE fixes https://github.com/aws-samples/aws-efa-eks/pull/21/files

File: patterns/istio/main.tf [hunk 5]
```diff
@@ -52,18 +52,16 @@ locals {
...
   cluster_name                   = local.name
-  cluster_version                = "1.28"
+  cluster_version                = "1.29"
   cluster_endpoint_public_access = true
 
   cluster_addons = {
     coredns    = {}
     kube-proxy = {}
-    vpc-cni = {
-      preserve = true
```
Comment (bryantbiggs) [inline patterns/istio/main.tf:65]: EKS module v20 now sets this by default

File: patterns/multi-tenancy-with-teams/main.tf [hunk 6]
```diff
@@ -50,10 +50,10 @@ locals {
 
 module "eks" {
   source  = "terraform-aws-modules/eks/aws"
-  version = "~> 19.16"
+  version = "~> 19.21"
```
Comment (bryantbiggs) [inline patterns/multi-tenancy-with-teams/main.tf:53]: need to update teams module to add native access entry support and then we can remove the `aws-auth` bits here and update to v20

File: patterns/privatelink-access/eks.tf [hunk 7]
```diff
@@ -16,10 +16,10 @@ provider "kubernetes" {
 
 module "eks" {
   source  = "terraform-aws-modules/eks/aws"
-  version = "~> 19.16"
+  version = "~> 19.21"
```
Comment (bryantbiggs) [inline patterns/privatelink-access/eks.tf:19]: will update in separate PR to resolve #1816

**Review by askulkarni2 — APPROVED** [review]

**Pull request merged**